skrub/_agg_joiner.py

```diff
@@ -6,57 +6,138 @@
-CATEG_OPERATIONS = ["mode", "count", "value_counts"]
-ALL_OPS = NUM_OPERATIONS + CATEG_OPERATIONS
+try:
+    import polars as pl
+except ImportError:
+    pass
+
+SUPPORTED_OPS = ["count", "mode", "min", "max", "sum", "median", "mean", "std"]
+# summing strings works in pandas, not in polars
+NUM_ONLY_OPS = ["sum", "median", "mean", "std"]
 
 
-def split_num_categ_operations(operations):
-    """Separate aggregator operators input by their type.
+def aggregate(table, key, cols_to_agg, operations, suffix):
+    """Aggregate the `table` by `key` and compute statistics for `cols_to_agg`.
 
     Parameters
     ----------
-    operations : list of str
-        The input operators names.
+    table : DataFrame
+        The input dataframe to aggregate.
+
+    key : iterable of str
+        The columns used as keys to aggregate on.
+
+    cols_to_agg : iterable of str
+        The columns to aggregate.
+
+    operations : iterable of str
+        The reduction functions to apply on columns in ``cols_to_agg``
+        during the aggregation.
+
+        Supported operations are "count", "mode", "min", "max", "sum", "median",
+        "mean", "std". The operations "sum", "median", "mean", "std" are reserved
+        to numeric type columns.
+
+    suffix : str
+        The suffix appended to output columns. Will only be applied
+        to columns created by the aggregations.
 
     Returns
     -------
-    num_operations, categ_operations : Tuple of list of str
-        List of operator names.
+    DataFrame
+        The aggregated output.
     """
-    num_operations, categ_operations = [], []
-    for operation in operations:
-        # hist(5) -> hist
-        op_root, _ = _parse_argument(operation)
-        if op_root in NUM_OPERATIONS:
-            num_operations.append(operation)
-        if op_root in CATEG_OPERATIONS:
-            categ_operations.append(operation)
-        if op_root not in ALL_OPS:
-            raise ValueError(f"operations options are {ALL_OPS}, got: {operation=!r}.")
+    table_to_agg = s.select(table, s.make_selector(key) | s.make_selector(cols_to_agg))
+
+    # Don't check the ID column, as it's not the one we aggregate on
+    table_to_check = s.select(table_to_agg, ~s.cols(*key))
+    not_numeric_cols = (~s.numeric()).expand(table_to_check)
+
+    num_only_op = list(set(operations).intersection(set(NUM_ONLY_OPS)))
+
+    if (len(not_numeric_cols) > 0) & (len(num_only_op) > 0):
+        raise AttributeError(
+            f"The operations {NUM_ONLY_OPS} are restricted to numeric columns."
+            f" \nConsider removing the following columns: {not_numeric_cols} or the"
+            f" following operations: {num_only_op}."
+        )
+
+    aggregated = perform_groupby(table_to_agg, key, cols_to_agg, operations)
+
+    new_col_names = [
+        f"{col}{suffix}" if col not in key else col
+        for col in sbd.column_names(aggregated)
+    ]
+    new_col_names = _join_utils.pick_column_names(new_col_names)
```

Comment:
why can there be some duplicates in the output column names? and if it can happen, shouldn't the deduplication be done when picking the `named_agg` / aliases?

Reply:
It could happen in the event where the main dataframe already has column names in the forme of `{col}_{operation}` like `rating_mean`, but I realize now it's handled in `left_join`. I don't believe it's actually possible there -- removing :)